fineract-provider/src/main/java/org/apache/fineract/portfolio/search/service/SearchReadPlatformServiceImpl.java

```diff
@@ -108,33 +108,33 @@ public String searchSchema(final SearchConditions searchConditions) {
 
         final String groupMatchSql = " (select CASE WHEN g.level_id=1 THEN 'CENTER' ELSE 'GROUP' END as entityType, g.id as entityId, g.display_name as entityName, g.external_id as entityExternalId, g.account_no as entityAccountNo, "
                 + " g.office_id as parentId, o.name as parentName, null as entityMobileNo, g.status_enum as entityStatusEnum, null as subEntityType, null as parentType "
-                + " from m_group g join m_office o on o.id = g.office_id where o.hierarchy like :hierarchy and (g.account_no like :search or g.display_name like :search or g.external_id like :search or g.id like :search )) "
+                + " from m_group g join m_office o on o.id = g.office_id where o.hierarchy like :hierarchy and (g.account_no like :search or g.display_name like :search or g.external_id like :search )) "
                 + " order by g.id desc";
 
         final StringBuilder sql = new StringBuilder();
 
         if (searchConditions.isClientSearch()) {
-            sql.append(clientMatchSql).append(union);
+            sql.append("(" + clientMatchSql + ")").append(union);
         }
 
         if (searchConditions.isLoanSeach()) {
-            sql.append(loanMatchSql).append(union);
+            sql.append("(" + loanMatchSql + ")").append(union);
         }
 
         if (searchConditions.isSavingSeach()) {
-            sql.append(savingMatchSql).append(union);
+            sql.append("(" + savingMatchSql + ")").append(union);
         }
 
         if (searchConditions.isShareSeach()) {
-            sql.append(shareMatchSql).append(union);
+            sql.append("(" + shareMatchSql + ")").append(union);
         }
 
         if (searchConditions.isClientIdentifierSearch()) {
-            sql.append(clientIdentifierMatchSql).append(union);
+            sql.append("(" + clientIdentifierMatchSql + ")").append(union);
```

Comment:
Please remove string concatenation by + operator... use the `append` method since you are using StringBuilder!